packages/multichain-account-service/src/providers/ProviderWrapper.ts

```diff
@@ -0,0 +1,129 @@
+import type { AccountProvider, Bip44Account } from '@metamask/account-api';
+import type { EntropySourceId, KeyringAccount } from '@metamask/keyring-api';
+
+/**
+ * A simple wrapper that adds disable functionality to any AccountProvider.
+ * When disabled, the provider will not create new accounts and return empty results.
+ */
+export class ProviderWrapper
```

Comment:
For consistency, I would also add a generic here and like `<Account extends KeyringAccount>` and re-use it in the other types that need one (thus we could use different account type if really needed).